components/isceobj/Sensor/UAVSAR_Stack.py

```diff
@@ -316,9 +312,9 @@ def extractDoppler(self):
         drho = instrument.getRangePixelSize()  #full res value, not spacing in the dop file
         prf = instrument.getPulseRepetitionFrequency()
         self.logger.info("extractDoppler: rho0, drho, prf = {}, {}, {}".format(rho0, drho, prf))
-        dopfile = self.metadata['dop']
-        f = open(dopfile,'r')
-        x = f.readlines()  #first line is a header
+        dopfile = self.dopplerFile if hasattr(self, 'dopplerFile') else self.metadata['dop']
```

Comment:
instead of hasattr, you can use getattr with the default value fallback

Reply:
Done. Thanks for the tip!